forem/forem
# Add Feeds::Import service class

## What type of PR is this? (check all applicable)


## Description

This PR is one step in a multi step plan which aims to find the best strategy to ingest XML (RSS, ATOM and other) feeds, parse them, transform them into articles to facilitate cross posting. 

Please read https://github.com/forem/forem/issues/10996 for the details and the discussion

## TODO

- [ ] add metrics if possible so that we can keep track on how each segment takes
- [ ] inline todos might be tackled in this or separate PRs

## QA Instructions, Screenshots, Recordings

1. delete all articles if possible
1. run `rails measure_feeds:setup` which adds a feed URL to each user you have locally
1. run `rails fetch_feeds_import`
1. check articles were imported

## Added tests?


## Added to documentation?
